File: example/src/main/kotlin/com/example/_01_creational_patterns/_02_factory_method/java/_03_java/SimpleFactory.java
package com.example._01_creational_patterns._02_factory_method.java._03_java;

import com.example._01_creational_patterns._02_factory_method.java._02_after.BlackShip;
import com.example._01_creational_patterns._02_factory_method.java._02_after.WhiteShip;

public class SimpleFactory {

    public Object createProduct(String name) {
        if (name.equals("whiteship")) {
            return new WhiteShip();
        } else if (name.equals("blackship")) {
            return new BlackShip();
        }

        throw new IllegalArgumentException();
    }
}

File: example/src/main/kotlin/com/example/_01_creational_patterns/_02_factory_method/kt/_03_java/SimpleFactory.kt
package com.example._01_creational_patterns._02_factory_method.kt._03_java

import com.example._01_creational_patterns._02_factory_method.java._02_after.BlackShip
import com.example._01_creational_patterns._02_factory_method.java._02_after.WhiteShip

class SimpleFactory {

    fun createProduct(name: String): Any {
        if (name == "whiteship") {
            return WhiteShip()
        } else if (name == "blackship") {
            return BlackShip()
        }
        throw IllegalArgumentException()
    }
}
